### DISCUSSION TITLE
After deploying to the vercel, my remix app is only returning a white screen, even though it is working fine on localhost.

### DISCUSSION THREAD

**Discussion by alikhantareen** [remix-run/remix#6092] [Q&A]
I have deployed the remix app on vercel and it was working fine. Recently, a couple of days ago, I pushed some changes to the main and the app hosted on to the vercel is only returning a white screen now. Routes are not working either. While, everything is working fine on localhost.

**Comment by kiliman** [comment]
What errors are shown in the dev console? Also are you sure you have received the initial SSR page (HTML) from the server?

**Comment by dgurns** [comment]
I just upgraded to remix `1.15.0` and am running into the same issue when deploying to Vercel. Tried a bunch of different permutations and configs, but consistently getting a 500 error which is caught in the root error boundary. 

```Failed to load resource: the server responded with a status of 500 ()```

<img width="524" alt="Screenshot 2023-04-22 at 17 30 36" src="https://user-images.githubusercontent.com/1173791/233796027-aac20b9a-c9d3-4c52-88f7-75172fa921ea.png">

Everything works fine locally in dev.

Here is my repo where you can see the full config: https://github.com/dgurns/trad-archive/tree/dpg-upgrade-deps/remix

**Reply by dgurns** [reply]
FYI, in case this is helpful for anyone in the future, I solved it with this config:

```js
// remix.config.js

/**
 * @type {import('@remix-run/dev').AppConfig}
 */
module.exports = {
	serverBuildPath: "api/index.js",
	serverMainFields: ["main", "module"],
	serverModuleFormat: "cjs",
	serverPlatform: "node",
	serverMinify: false,
	ignoredRouteFiles: ["**/.*"],
	// When running locally in development mode, we use the built-in remix
	// server. This does not understand the vercel lambda module format,
	// so we default back to the standard build output.
	server: process.env.NODE_ENV === "development" ? undefined : "./server.ts",
	future: {
		v2_errorBoundary: true,
		v2_normalizeFormMethod: true,
		v2_meta: true,
		v2_routeConvention: true,
	},
};
```

```ts
// entry.client.tsx

import { RemixBrowser } from "@remix-run/react";
import { hydrate } from "react-dom";

hydrate(<RemixBrowser />, document);
```

```ts
// entry.server.tsx

import { type EntryContext } from "@remix-run/node";
import { RemixServer } from "@remix-run/react";
import { renderToString } from "react-dom/server";
import etag from "etag";

export default function handleRequest(
	request: Request,
	responseStatusCode: number,
	responseHeaders: Headers,
	remixContext: EntryContext
) {
	let markup = renderToString(
		<RemixServer context={remixContext} url={request.url} />
	);
	const generatedEtag = etag(markup);
	responseHeaders.set("Content-Type", "text/html");
	responseHeaders.set("ETag", generatedEtag);

	// check if the `If-None-Match` header matches the ETag; if so, send a 304 so
	// the browser can use its cached document
	if (request.headers.get("If-None-Match") === responseHeaders.get("ETag")) {
		return new Response("", { status: 304, headers: responseHeaders });
	}

	return new Response("<!DOCTYPE html>" + markup, {
		status: responseStatusCode,
		headers: responseHeaders,
	});
}
```

**Reply by lili21** [reply]
out of curious, is it worth to do a etag compare for SSR?

**Reply by WesleyYue** [reply]
I'm also suddenly seeing this issue and redeployments of previously working deployments stopped working. What specifically do you think is the problem? I can't copy paste what you have here since we have some custom stuff on streaming to make emotion work
